Source/WebCore/rendering/RenderThemeIOS.mm

```diff
@@ -757,7 +764,12 @@ static void applyCommonButtonPaddingToStyle(RenderStyle& style, const Element& e
     auto emSize = CSSPrimitiveValue::create(0.5, CSSUnitType::CSS_EM);
     // We don't need this element's parent style to calculate `em` units, so it's okay to pass nullptr for it here.
     int pixels = emSize->computeLength<int>({ style, document.renderStyle(), nullptr, document.renderView() });
-    style.setPaddingBox(LengthBox(0, pixels, 0, pixels));
+
+    auto paddingBox = LengthBox(0, pixels, 0, pixels);
+    if (!style.isHorizontalWritingMode())
+        paddingBox = LengthBox(paddingBox.left().value(), paddingBox.top().value(), paddingBox.right().value(), paddingBox.bottom().value());
```

Comment:
Starting to think we should have a helper for this, can be a follow up though.

Reply:
Agreed.